pkg/user/service.go

```diff
@@ -65,11 +66,14 @@ func hashPassword(password string) (string, error) {
 }
 
 func (s service) SignIn(email string, password string) (*model.User, error) {
-	unauthorizedMessage := "invalid email and password combination"
+	unauthorizedError := fmt.Errorf("invalid email and password combination")
 
 	user, err := s.repository.findByEmail(email)
 	if err != nil {
-		return nil, errors.New(unauthorizedMessage)
+		if errdef.IsNotFound(err) {
+			return nil, errdef.NewUnauthorized(unauthorizedError)
+		}
+		return nil, err
```

Comment:
Should it only be Unauthorized if we cannot find the user? I thought the error code should stay the same, regardless of its root. At least for these auth errors.